synapse/storage/data_stores/main/state.py

```diff
@@ -285,7 +285,9 @@ def get_room_predecessor(self, room_id):
             room_id (str)
 
         Returns:
-            Deferred[unicode|None]: predecessor room id
+            Deferred[dict[str, str]|None]: A dictionary containing the following keys:
```

Comment:
hrm. I think it's worth saying that it contains the content of the `predecessor` field from the create event. You *hope* that has the structure described, but particularly in the case of federated rooms, you have no guarantee of that at all.